src/preact/slot.js

```diff
@@ -59,14 +59,16 @@ export function Slot(props) {
 export function useSlotContext(ref) {
   const context = useAmpContext();
   useLayoutEffect(() => {
-    const slot = dev().assertElement(ref.current);
+    const slot = ref.current;
+    devAssert(slot?.nodeType == 1, 'Element expected');
+
     setProp(slot, CanRender, Slot, context.renderable);
     setProp(slot, CanPlay, Slot, context.playable);
     setProp(
       slot,
       LoadingProp,
       Slot,
-      /** @type {!../loading.Loading} */ (context.loading)
+      /** @type {!./loading-instructions.Loading} */ (context.loading)
```

Comment:
Same with `loading.js`. Can we move it to `src/pure`? It's already designed to be so. But it's not directly a "Preact" thing.

Reply:
Moved it into `src/core`. Referring to comments above, `src/context` is already whitelisted, while `src/contextprops` was moved into `src/preact`

Reply:
`contextprops` is not really a "Preact" thing. It just maps well into it and consistent. But it's meant to go beyond into the AMP Layer.

Reply:
My bad; it looked very Preact-related. Moved into `src/core`

Reply:
Thanks! And, no prob, of course :)

Reply:
```suggestion
      /** @type {!../core/loading-instructions.Loading} */ (context.loading)
```